Buildings/Controls/OBC/ASHRAE/G36/Atomic/SupplyAirTempSet_MultiZone.mo

```diff
@@ -0,0 +1,330 @@
+//   parameter Integer numReq = cooReq
+//     "Number of cooling requests"
+//     annotation(Dialog(group="Variables of Trim & Response logic to set maximum supply temperature T_max"));
+//   parameter Modelica.SIunits.Temperature triAmo = 0.1
+//     "Trim amount"
+//     annotation (Dialog(group="Variables of Trim & Response logic to set maximum supply temperature T_max"));
+//   parameter Modelica.SIunits.Temperature resAmo = -0.2
+//     "Response amount"
+//     annotation (Dialog(group="Variables of Trim & Response logic to set maximum supply temperature T_max"));
+//   parameter Modelica.SIunits.Temperature maxRes = -0.6
+//     "Maximum response per time interval"
+//     annotation (Dialog(group="Variables of Trim & Response logic to set maximum supply temperature T_max"));
+
+protected
+  CDL.Continuous.Line lin
+  "Supply temperature distributes linearly between TSupMin and TSupMax, according to Tout"
+    annotation (Placement(transformation(extent={{0,40},{20,60}})));
+  CDL.Continuous.Sources.Constant minOutTem(k=TOutMin)
+    "Lower value of the outdoor air temperature reset range"
+    annotation (Placement(transformation(extent={{-60,60},{-40,80}})));
+  CDL.Continuous.Sources.Constant maxOutTem(k=TOutMax)
+    "Higher value of the outdoor air temperature reset range"
+    annotation (Placement(transformation(extent={{-60,20},{-40,40}})));
+  CDL.Continuous.Sources.Constant minSupTem(k=TSupMin)
+    "Lowest cooling supply air temperature setpoint"
+    annotation (Placement(transformation(extent={{-60,-20},{-40,0}})));
+  CDL.Logical.And and2 "Check if it is in Setup or Cool-down mode"
+    annotation (Placement(transformation(extent={{0,-60},{20,-40}})));
+  CDL.Logical.And and1 "Check if it is in Warmup or Setback mode"
+    annotation (Placement(transformation(extent={{40,-100},{60,-80}})));
+  CDL.Continuous.Sources.Constant TSupWarUpSetBac(k=35 + 273.15)
+    "Supply temperature setpoint under warm-up and setback mode"
+    annotation (Placement(transformation(extent={{40,-140},{60,-120}})));
+  CDL.Logical.Switch swi1
+    "If operation mode is setup or cool-down, setpoint shall be 35 degC"
+    annotation (Placement(transformation(extent={{100,-40},{120,-60}})));
+  CDL.Logical.Switch swi2
+    "If operation mode is setup or cool-down, setpoint shall be TSupMin"
+    annotation (Placement(transformation(extent={{40,-60},{60,-40}})));
+  CDL.Continuous.Limiter TDea(
+    uMax=24 + 273.15,
+    uMin=21 + 273.15)
+    "Limiter that outputs the dead band value for the supply air temperature"
+    annotation (Placement(transformation(extent={{40,80},{60,100}})));
+  CDL.Logical.Switch swi3 "Check output regarding supply fan status"
+    annotation (Placement(transformation(extent={{100,20},{120,0}})));
+  CDL.Integers.LessThreshold intLesThr(threshold=Constants.OperationModes.warUpInd)
+    "Check if operation mode index is less than warm-up mode index (4)"
```

Comment:
It is better to not state the numerical value. Otherwise, if we ever need to change `Constants.OperationModes`, the comment will be wrong, and we probably won't detect it